### PR TITLE
Optimization of the first call of zim::Archive::iterEfficient()

### PR DESCRIPTION
**Pull request by veloman-yunkan**
1. memory usage of `FileImpl::m_articleListByCluster` has been halved
2. general purpose O(NlogN) sorting algorithm was replaced with a counting sort approach (with O(N) time complexity).

### CODE REVIEW

File: src/fileimpl.cpp [hunk 1]
```diff
@@ -350,22 +397,24 @@ makeFileReader(std::shared_ptr<const FileCompound> zimFile, offset_t offset, zsi
   {
     m_articleListByCluster.reserve(getUserEntryCount().v);
```
Comment (mgautierfr) [inline src/fileimpl.cpp:429]: We don't need this reserve now

**Review by mgautierfr — CHANGES_REQUESTED** [review]
A small change needed but it seems good to me.
Nice improvement !

File: src/fileimpl.cpp (hunk 1 above)
Comment (veloman-yunkan) [reply src/fileimpl.cpp]: Good catch! Fixed.

**Comment by mgautierfr** [comment]
LGTM. Please rebase-fixup on master

**Comment by veloman-yunkan** [comment]
> LGTM. Please rebase-fixup on master

Done

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**